test/extensions/common/wasm/wasm_test.cc

```diff
@@ -1399,6 +1399,91 @@ TEST_P(WasmCommonContextTest, EmptyContext) {
   root_context_->validateConfiguration("", plugin_);
 }
 
+class TestFilter : public Envoy::Extensions::Common::Wasm::Context {
```

Comment:
Don't we already have this kind of class for tests? eg shouldn't what we have for `ROOT_FACTORY(EnvoyRootContext), "empty")` be reused?

Reply:
Did you mean here?  I have changed to reuse the `EnvoyRootContext`:
https://github.com/envoyproxy/envoy/pull/23049/files#diff-fb64400ce23fde0ad1cad5c6c7fb205733c5b39c540e4687547fcda36971941aR114

Reply:
no I mean I think you shouldn't need to definite this class in the first place. You should be able to reuse `WasmCommonContextTest` and control the in-Wasm root context by simply passing "send local reply twice" to the setUp method. See https://github.com/envoyproxy/envoy/blob/c94ee6c23b75591ebd54ef29b48aa82792be41ba/test/extensions/common/wasm/wasm_test.cc#L1382-L1400

Reply:
or at least modify that `WasmCommonContextTest` class to reuse it here and "empty" test as well

Reply:
done

Reply:
Cool looking good